AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -248,6 +249,25 @@ - (void)fetchData
 
 #pragma mark - Private methods -- only call with lock.
 
+- (void)_clearImage
+{
+  // Destruction of bigger images on the main thread can be expensive
+  // and can take some time, so we dispatch onto a bg queue to
+  // actually dealloc.
+  UIImage *image = self.image;
+  CGSize imageSize = image.size;
+  BOOL shouldReleaseImageOnBackgroundThread = imageSize.width > kMinReleaseImageOnBackgroundSize.width ||
+                                              imageSize.height > kMinReleaseImageOnBackgroundSize.height;
+  if (shouldReleaseImageOnBackgroundThread) {
+    __block UIImage *blockImage = image;
```

Comment:
Technically we should not need __block here, as we don't need to write to the variable but just have it go out of scope.   simply referencing the image with (__unused) or something should be enough - but this is fine too! :)